src/main/java/com/avispl/dal/communicator/cisco/CiscoCommunicator.java

```diff
@@ -597,6 +597,123 @@ public void unmute() throws Exception {
             doPost("putxml", muteCommand);
         }
 
+    /**
+     * Route media channel data between the legacy firmware way and the CE9+ way
+     *
+     * @param ciscoStatus to get status information from
+     * @param endpointStatistics to put media channel data to
+     *
+     * @since 1.1.0
+     */
+    private void routeMediaChannelsData(CiscoStatus ciscoStatus, EndpointStatistics endpointStatistics) {
+            MediaChannels mediaChannels = ciscoStatus.getMediaChannels();
+
+            if (mediaChannels == null) {
+                // this may mean that we are going legacy route since channel data is in the different part of payload in this case
+                populateCallChannelsData(ciscoStatus, endpointStatistics);
+            } else {
+                populateMediaChannelsData(ciscoStatus, endpointStatistics);
+            }
+        }
+
+    /**
+     * Populate media channels information, retrieved from the Call statistics of the device
+     *
+     * @param ciscoStatus to retrieve statistics from
+     * @param endpointStatistics to save data to
+     *
+     * @since 1.1.0
+     */
+    private void populateCallChannelsData(CiscoStatus ciscoStatus, EndpointStatistics endpointStatistics) {
+            Call[] calls = ciscoStatus.getCalls();
+            if (calls == null) {
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Unable to populate media channels data: no calls information is available");
+                }
+                return;
+            }
+
+            List<Call> connectedCalls = Arrays.stream(ciscoStatus.getCalls()).filter(call -> "Connected".equals(call.getStatus()) || "Synced".equals(call.getStatus()))
```

Comment:
no big deal, but you can write this line as follows to avoid extracting status twice :)
```
List<Call> connectedCalls = Arrays.stream(ciscoStatus.getCalls()).filter(call -> {
			String callStatus = call.getStatus();

			if ("Connected".equals(callStatus) || "Synced".equals(callStatus)) {
				return true;
			} else {
				return false;
			}
		}).collect(Collectors.toList());
```